tests/pages/RightPanelPageObject.ts

```diff
@@ -0,0 +1,267 @@
+      },
+      toggleLock: async () => {
+        await actionsButton.click();
+        await this.page.getByTestId('LockToggle').click();
+      },
+      unlock: async () => {
+        await actionsButton.click();
+        await this.page.getByTestId('Unlock').click();
+      },
+      rename: async (text: string) => {
+        await actionsButton.click();
+        await this.page.getByTestId('Rename').click();
+        await this.overlayPageObject.dialog.input.fillAndSave(text);
+      },
+    };
+  }
+
+  private getPanelRowDataObject(row: Locator) {
+    const getActionsMenu = (row: Locator) => this.getActionsMenu.bind(this)(row);
+
+    return {
+      get actions() {
+        return getActionsMenu(row);
+      },
+      click: async () => {
+        await row.click();
+      },
+      locator: row,
+      toggleVisibility: async () => {
+        await row.getByTestId('visibilityToggle').click();
+      },
+    };
+  }
+
+  private getPanelRowByIdx(index: number) {
+    const row = this.page.getByTestId('data-row').nth(index);
+    return this.getPanelRowDataObject(row);
+  }
+
+  private getPanelRowByText(text: string) {
+    const row = this.page.getByTestId('data-row').filter({ hasText: text });
+    return this.getPanelRowDataObject(row);
+  }
+
+  async toggle() {
+    await this.page.getByTestId('side-panel-header-right').click();
+  }
+
+  get measurementsPanel() {
+    const page = this.page;
+    const getMeasurementByIdx = (index: number) => this.getPanelRowByIdx(index);
+    const getMeasurementByText = (text: string) => this.getPanelRowByText(text);
+
+    return {
+      get menuButton() {
+        const button = page.getByTestId('trackedMeasurements-btn');
+        return {
+          button,
+          click: async () => {
+            await button.click();
+          },
+        };
+      },
+      panel: {
+        deleteAll: async () => {
+          await page.getByRole('button', { name: 'Delete' }).click();
+        },
+        getMeasurementCount: async () => {
+          return await page.getByTestId('data-row').count();
+        },
+        locator: page.getByTestId('trackedMeasurements-panel').last(),
+        nthMeasurement(index: number) {
+          return getMeasurementByIdx(index);
+        },
+        measurementByText(text: string) {
+          return getMeasurementByText(text);
+        },
+      },
+    };
+  }
+
+  get segmentationPanel() {
+    const page = this.page;
+    const getSegmentationByIdx = (index: number) => this.getPanelRowByIdx(index);
+    const getSegmentationByText = (text: string) => this.getPanelRowByText(text);
+
+    return {
+      get addSegmentationButton() {
+        const button = page.getByTestId('addSegmentation');
+        return {
+          button,
+          click: async () => {
+            await button.click();
+          },
+        };
+      },
+      get contourMenuButton() {
+        const button = page.getByTestId('panelSegmentationWithToolsContour-btn');
```

Comment:
This button and the labelMapMenuButton below are NOT part of the segmentation panel, but instead are the tabs/buttons to change what is displayed in the right panel. So please move these accessors directly into the RightPanelPageObject. Or maybe just rename them panelTabButton to make it clear these are the buttons to switch to that panel.

Reply:
Done at 6644c97e6578b33bfaa4e1aaec10245338b1192c